### ISSUE TITLE
Users are allowed to create multiple accounts under one email with current sign up route.

### ISSUE THREAD

**Issue by ropeace8** [Tech-Nc-State/tech-at-hub#83]
Currently during sign up in the backend (UserController.cs) we check if a user with the entered username already exists in the DB: `User? existingUser = _dbContext.Users.Where(u => u.Username == `form.Username).FirstOrDefault();`

With this logic multiple accounts could be created using the same email, which is a potential vulnerability when thinking about DDOS type attacks.

We should probably expand this check to include email.

**Comment by Heckin-Doggo** [comment]
yeah this just straight up crashes the backend. We should probably fix it.

**Comment by jpbream** [comment]
Yep, lets add a unique constraint on the email
